### PR TITLE
config: rename configuration variable according to conventions

### LINKED ISSUE #2785: global: rename all CFG configuration variables
Configuration variables starting with `CFG` are a relic from the past, and don't satisfy the Flask convention for configuration variable names, leading to potential clashes and obscure names (next year who is going to know what `CFG_BIBCATALOG` is?)

In fact, configuration variables relevant only inside module `foo` should start with `FOO_` ([example](https://github.com/inspirehep/inspire-next/blob/eb08c999b2190a2c64f392623df63deaee97068c/inspirehep/modules/hal/config.py)), while configuration variables used all across INSPIRE should start with `INSPIRE_`.

### PR DESCRIPTION
**Pull request by szymonlopaciuk**
## Description

- rename `CFG_*` configuration variables
- make variables conform to the convention `MODULENAME_*` or `INSPIRE_*`

Each commit contains a summary of all the changes in its message.

## Related Issue
fixes #2785 

## Checklist:

- [ ] I have all the information that I need (if not, move to `RFC` and look for it).
- [ ] I linked the related issue(s) in the corresponding commit logs.
- [x] I wrote [good commit log messages](https://github.com/torvalds/subsurface-for-dirk/blob/5f15ad5a86ada3c5e574041a5f9d85235322dabb/README#L92-L119).
- [x] My code follows the code style of this project.
- [x] I've added any new docs if API/utils methods were added.
- [x] I have updated the existing documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: inspirehep/utils/ext.py [hunk 1]
```diff
@@ -44,14 +44,14 @@ def init_app(self, app):
...
+        url = app.config.get("UTILS_BIBCATALOG_REQUEST_TRACKER_URL", "")
+        login = app.config.get("UTILS_BIBCATALOG_REQUEST_TRACKER_DEFAULT_USER", "")
         verify_cert = app.config.get(
-            "CFG_BIBCATALOG_SYSTEM_RT_VERIFY_SSL",
+            "UTILS_BIBCATALOG_REQUEST_TRACKER_VERIFY_SSL",
             True,
         )
         password = app.config.get(
-            "CFG_BIBCATALOG_SYSTEM_RT_DEFAULT_PWD", "")
+            "UTILS_BIBCATALOG_REQUEST_TRACKER_DEFAULT_PWD", "")
```
Comment (jacquerie) [inline inspirehep/utils/ext.py]: I think we can omit `BIBCATALOG` everywhere.

File: tests/unit/workflows/test_workflows_tasks_matching.py [hunk 2]
```diff
@@ -141,9 +141,9 @@ def test_belongs_to_relevant_category_returns_false(app):
     subschema = schema['properties']['arxiv_eprints']
 
     categories_config = {
-        'ARXIV_CATEGORIES': {
+        'WORKFLOWS_ARXIV_CATEGORIES': {
             'core':     ['hep-ph'],
-            'non-core': ['astro-ph.CO', 'gr-qc']
+            'non-core': ['astro-ph.CWORKFLOWS_O', 'gr-qc']
```
Comment (jacquerie) [inline tests/unit/workflows/test_workflows_tasks_matching.py]: typo : )

File: inspirehep/config.py [hunk 3]
```diff
@@ -1408,173 +1410,11 @@
...
-    "solv-int": "Math and Math Physics",
-    "stat": "Other",
-    "stat.AP": "Other",
-    "stat.CO": "Other",
-    "stat.ME": "Other",
-    "stat.ML": "Other",
-    "stat.OT": "Other",
-    "stat.TH": "Other"
-}
+ARXIV_TO_INSPIRE_CATEGORY_MAPPING = schema_utils.ARXIV_TO_INSPIRE_CATEGORY_MAPPING
```
Comment (jacquerie) [inline inspirehep/config.py]: You don't even need this: importing the name creates the binding, so just doing `from inspire_schemas.utils import ARXIV_TO_INSPIRE_CATEGORY_MAPPING` is enough.

**Comment by ammirate** [comment]
What's missing for this PR to be ready?

**Comment by szymonlopaciuk** [comment]
@ammirate I renamed everything I knew what to do with, so nothing from me I think, unless there are any suggestions. I know @jacquerie mentioned merging will be tricky, as these changes imply changes to config files.

**Comment by jacquerie** [comment]
Stale, will be resurrected if/when needed.

**Pull request closed without merging**